### PR TITLE
mcs: use getClusterInfo to check whether api service is ready

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5836

It is a pick from pd-cse.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by binshi-bing — APPROVED** [review]
LGTM

File: tests/integrations/mcs/testutil.go [hunk 1]
```diff
@@ -85,19 +85,22 @@ func StartSingleResourceManagerTestServer(ctx context.Context, re *require.Asser
 	return s, cleanup
 }
 
-// StartSingleTSOTestServer creates and starts a tso server with default config for testing.
-func StartSingleTSOTestServer(ctx context.Context, re *require.Assertions, backendEndpoints, listenAddrs string) (*tso.Server, func()) {
+// StartSingleTSOTestServerWithoutCheck creates and starts a tso server with default config for testing.
+func StartSingleTSOTestServerWithoutCheck(ctx context.Context, re *require.Assertions, backendEndpoints, listenAddrs string) (*tso.Server, func(), error) {
```
Comment (rleungx) [inline tests/integrations/mcs/testutil.go:89]: What does the WithoutCheck mean?

**Review by rleungx — APPROVED** [review]
The rest LGTM

File: tests/integrations/mcs/testutil.go (hunk 1 above)
Comment (lhy1024) [reply tests/integrations/mcs/testutil.go:89]: without check error and primary

**Pull request merged**